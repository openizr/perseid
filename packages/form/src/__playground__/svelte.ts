/* c8 ignore start */
/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call */

import { SvelteComponent } from 'svelte';
import Engine from 'scripts/core/Engine';
import Form from 'scripts/svelte/Form.svelte';
import Field from '__playground__/Field.svelte';

let app: Form;
const { log } = console;

function main(): void {
  app = new Form({
    props: {
      Field: Field as typeof SvelteComponent | null | undefined,
      configuration: {
        root: 'root',
        fields: {
          test: { type: 'string', required: true },
        },
        steps: {
          root: {
            fields: ['test'],
          },
        },
        plugins: [
          (engine: Engine): void => {
            engine.on('userAction', (data, next) => {
              log(data);
              return next(data);
            });
          },
        ],
      },
    },
    target: document.getElementById('root') as unknown as HTMLElement,
  });
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { $destroy: () => void; }).$destroy();
});
