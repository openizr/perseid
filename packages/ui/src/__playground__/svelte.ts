/* c8 ignore start */

import type { SvelteComponent } from 'svelte';
import Router from '__playground__/pages/Router.svelte';

let app: SvelteComponent;

function main(): void {
  const target = document.querySelector('#root') as unknown as HTMLElement;
  target.innerHTML = '';
  app = new Router({
    hydrate: false,
    target,
  });
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Svelte components tree.
window.addEventListener('beforeunload', () => {
  app.$destroy();
});
