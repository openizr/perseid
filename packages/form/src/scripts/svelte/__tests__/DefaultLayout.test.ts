/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { type SvelteComponent } from 'svelte';
import { render } from '@testing-library/svelte';
import DefaultLayout from 'scripts/svelte/DefaultLayout.svelte';
import DefaultLoader from 'scripts/svelte/DefaultLoader.svelte';

describe('svelte/DefaultLayout', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly', () => {
    const { container } = render(DefaultLayout, {
      props: {
        activeStep: 'root',
        Loader: DefaultLoader as unknown as typeof SvelteComponent,
        setActiveStep: vi.fn(),
        useSubscription: vi.fn(),
        state: {
          steps: [],
          loading: true,
          variables: {},
          userInputs: { full: {}, partial: {} },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
