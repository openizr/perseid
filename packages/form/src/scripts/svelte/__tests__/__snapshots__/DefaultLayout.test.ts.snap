// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`svelte/DefaultLayout > renders correctly 1`] = `
<div>
  <div
    class="perseid-form__steps"
  >
     
    <!--&lt;DefaultLoader&gt;-->
    
  </div>
  <!--&lt;DefaultLayout&gt;-->
</div>
`;
